### ISSUE TITLE
[Bug]: Hangs at creating collection after Milvus recovered from datanode pod kill chaos

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#17366]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20220602-48cf63a2
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus==2.1.0.dev66
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

run e2e test after chaos timeout
```
testcases/test_e2e.py::TestE2e::test_milvus_default 
-------------------------------- live log setup --------------------------------
[2022-06-04 18:50:33 - INFO - ci_test]: ################################################################################ (conftest.py:168)
[2022-06-04 18:50:33 - INFO - ci_test]: [initialize_milvus] Log cleaned up, start testing... (conftest.py:169)
[2022-06-04 18:50:33 - INFO - ci_test]: [setup_class] Start setup class... (client_base.py:30)
[2022-06-04 18:50:33 - INFO - ci_test]: *********************************** setup *********************************** (client_base.py:36)
[2022-06-04 18:50:33 - INFO - ci_test]: [setup_method] Start setup test case test_milvus_default. (client_base.py:37)
Error: The action has timed out.
```
from the log in proxy, after the request to create a collection is accepted, there is no subsequent log. We can speculate that it hangs at the step of  creating the collection

### Expected Behavior

all test cases passed

### Steps To Reproduce

```markdown
see https://github.com/milvus-io/milvus/runs/6740010092?check_suite_focus=true
```

### Milvus Log

failed job: https://github.com/milvus-io/milvus/runs/6740010092?check_suite_focus=true
log: https://github.com/milvus-io/milvus/suites/6794766675/artifacts/260723042

**Comment by soothing-rain** [comment]
@yhmo can you take a look at this?

**Comment by zhuwenxing** [comment]
failed job: https://github.com/milvus-io/milvus/runs/6800569627?check_suite_focus=true
log: https://github.com/milvus-io/milvus/suites/6850926209/artifacts/264489120

log in proxy stopped at `CreateCollection enqueued`
```
[2022/06/08 19:48:25.602 +00:00] [DEBUG] [impl.go:361] ["HasCollection received"] [traceID=66fd29d4f9d1fe8d] [role=proxy] [db=] [collection=e2e__YKWTejCa]
[2022/06/08 19:48:25.605 +00:00] [DEBUG] [id.go:140] ["IDAllocator pickCanDoFunc"] [need=1] [total=156097] [remainReqCnt=0]
[2022/06/08 19:48:25.606 +00:00] [DEBUG] [impl.go:392] ["HasCollection enqueued"] [traceID=66fd29d4f9d1fe8d] [role=proxy] [MsgID=433774211460541312] [BeginTS=433774318210252802] [EndTS=433774318210252802] [db=] [collection=e2e__YKWTejCa]
[2022/06/08 19:48:25.609 +00:00] [DEBUG] [impl.go:422] ["HasCollection done"] [traceID=66fd29d4f9d1fe8d] [role=proxy] [MsgID=433774211460541312] [BeginTS=433774318210252802] [EndTS=433774318210252802] [db=] [collection=e2e__YKWTejCa]
[2022/06/08 19:48:25.613 +00:00] [DEBUG] [impl.go:178] ["CreateCollection received"] [traceID=2a173295f307f181] [role=proxy] [db=] [collection=e2e__YKWTejCa] [len(schema)=103] [shards_num=2] [consistency_level=Strong]
[2022/06/08 19:48:25.614 +00:00] [DEBUG] [id.go:140] ["IDAllocator pickCanDoFunc"] [need=1] [total=156096] [remainReqCnt=0]
[2022/06/08 19:48:25.614 +00:00] [DEBUG] [impl.go:207] ["CreateCollection enqueued"] [traceID=2a173295f307f181] [role=proxy] [MsgID=433774211460541313] [BeginTs=43377431821864[140](https://github.com/milvus-io/milvus/runs/6800569627?check_suite_focus=true#step:13:141)9] [EndTs=433774318218641409] [timestamp=433774318218641409] [db=] [collection=e2e__YKWTejCa] [len(schema)=103] [shards_num=2] [consistency_level=Strong]
```

**Comment by zhuwenxing** [comment]
failed job:  http://10.100.32.144:8080/blue/organizations/jenkins/chaos-test-debug/detail/chaos-test-debug/41/pipeline/179/
log:  [artifacts-datanode-pod-kill-41-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/8867326/artifacts-datanode-pod-kill-41-server-logs.tar.gz)

**Comment by xiaofan-luan** [comment]
@yhmo any progress on it, any help needed?

**Comment by yhmo** [comment]
@xiaofan-luan I am working on bulkload problems, no time to look at this one currently.

**Comment by zhuwenxing** [comment]
so for the datanode pod kill, there will be the problem of not being able to create a collection, there are two behaviors, one is the current issue, and the other is https://github.com/milvus-io/milvus/issues/17335

**Comment by soothing-rain** [comment]
@zhuwenxing Looking at the log it seems DataNode kept restarting during the test? Can you find out why DataNode kept crashing (or getting killed)? From the logs it was quite chaotic :d

For example I saw `["received datanode register"]` 13 times on DatNodeID=7, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 

These logs look like:
```
[2022/06/09 04:10:31.904 +00:00] [INFO] [server.go:760] ["received datanode register"] [address=10.97.17.112:21124] [serverID=20]
```

However I only saw `["received datanode unregister"]` 10 times on DataNodeID=7, 9, 11, 12, 13, 14, 15, 16, 17, 18.

These logs look like:
```
[2022/06/09 04:10:26.589 +00:00] [INFO] [server.go:769] ["received datanode unregister"] [address=10.97.19.221:21124] [serverID=18]
```

Also looking at DataNode's log, it has nodeID=`36`:
```
[2022/06/09 04:11:53.874 +00:00] [INFO] [session_util.go:241] ["Service registered successfully"] [ServerName=datanode] [serverID=36]
```

So basically three questions here:
1. Why did DataNode keep crashing (or keep getting killed)?
2. Why are we not receiving datanode unregister message for node `10` and `19`. Even after 10 minutes (as it appears from the log)
3. What was going on with DataNode `21` to DataNode `35`?

**Comment by soothing-rain** [comment]
/unassign
/assign @zhuwenxing 

please verify

**Comment by zhuwenxing** [comment]
Not reproduced yet, so remove the critical label.
